RDBM/Record.cpp

```diff
@@ -0,0 +1,181 @@
+#include"Record.h"
+#include<cstdlib>
+Record::Record()
+{
+	id = ++ID;
+}
+
+Record& Record::Add(std::string value)
+{
+	char type;
+	int ival;
+	double dval;
+	int checki = 1;
+	int checkd1 = 1;
+	int checkd2 = 0;
+	for (int i = 0; i < value.size(); i++)
+	{
+		if (isdigit(value[i]))
+		{
+			checki *= 1;
+			checkd1 *= 1;
+		}
+		else if (value[i] == '.')
+		{
+			checkd2++;
+		}
+		else
+		{
+			checki = 0;
+			checkd1 = 0;
+		}
+		checki *= checki;
+		checkd1 *= checkd1;
+	}
+	
+	
+	if (checkd1&&checkd2 == 1)
+	{
+		type = 'd';
+		dval = atof(value.c_str());
+	}
+	else if (checki)
+	{
+		type = 'i';
+		ival = atoi(value.c_str());
+	}
+	else
+	{
+		type = 's';
+	}
+	switch (type)
+	{
+	case 'i':
+	{
+		Integer *cell = new Integer(ival);
+		record.push_back(cell);
+		break;
+	}
+	case 'd':
+	{
+				Double *cell = new Double(dval);
+		record.push_back(cell);
+		break;
+	}
+	case 's':
+	{
+		String *cell = new String(value);
+		record.push_back(cell);
+		break;
+	}
+	default:
+		break;
+	}
+	return *this;
+}
+
+void Record::Show(std::ostream &out)
+{
+	for (unsigned int i = 0; i < record.size(); i++)
+	{
+		record[i]->Show(out);
+	}
+	out << std::endl;
+}
+
+void Record::IdReset(int newid)
+{
+	id = newid;
+}
+
+Record& Record::Set(int index,std::istream &in)
+{
+	index--;
+	if (index < record.size()&&index>=0)
+	{
+		record[index]->Set(in);
+	}
+	return *this;
+}
+
+bool Record::Find(std::string value)
+{
+	char type;
+	int checki = 1;
+	int checkd1 = 1;
+	int checkd2 = 0;
+	for (int i = 0; i < value.size(); i++)
+	{
+		if (isdigit(value[i]))
+		{
+			checki *= 1;
+			checkd1 *= 1;
+		}
+		else if (value[i] == '.')
+		{
+			checkd2++;
+		}
+		else
+		{
+			checki = 0;
+			checkd1 = 0;
+		}
+		checki *= checki;
+		checkd1 *= checkd1;
+	}
+	if (checkd1&&checkd2 == 1)
```

Comment:
First of all i suggest you to think about better solution of parse this string.
Second - you have the same code as in Add function.
Move it to separate function.